ui/src/treegrid.tsx

```diff
@@ -0,0 +1,253 @@
+import ErrorIcon from "@mui/icons-material/Error";
+import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
+import KeyboardArrowRightIcon from "@mui/icons-material/KeyboardArrowRight";
+import CircularProgress from "@mui/material/CircularProgress";
+import IconButton from "@mui/material/IconButton";
+import Typography from "@mui/material/Typography";
+import { useCallback } from "react";
+import { useImmer } from "use-immer";
+
+export type TreeGridId = string | number;
+
+export type TreeGridRowData = {
+  [key: string]: string | number | JSX.Element;
+};
+
+export type TreeGridItem = {
+  children?: TreeGridId[];
+  data: TreeGridRowData;
+};
+
+export type TreeGridData = {
+  [key: TreeGridId]: TreeGridItem;
+};
+
+export type TreeGridColumnData = {
+  key: string;
+  width: string | number;
+  title?: string | JSX.Element;
+};
+
+export type TreeGridProps = {
+  columns: TreeGridColumnData[];
+  data: TreeGridData;
+  prefixElements?: (id: TreeGridId, data: TreeGridRowData) => JSX.Element[];
```

Comment:
shouldn't this just return ReactNode and then you can return any valid JSX?  as an example, its weird that your test has to return an array